src/shared/components/common/markdown-textarea.tsx

```diff
@@ -374,10 +374,59 @@ export class MarkdownTextArea extends Component<
     autosize.update(textarea);
   }
 
-  handleImageUploadPaste(i: MarkdownTextArea, event: any) {
+  handlePaste(i: MarkdownTextArea, event: ClipboardEvent) {
+    if (!event.clipboardData) return;
+
+    // check clipboard files
     const image = event.clipboardData.files[0];
     if (image) {
       i.handleImageUpload(i, image);
+      return;
+    }
+
+    // check clipboard url
+    const url = i.isValidUrl(event.clipboardData.getData("text"));
+    if (url) {
+      i.handleUrlPaste(url, i, event);
+      return;
+    }
+  }
+
+  handleUrlPaste(url: URL, i: MarkdownTextArea, event: ClipboardEvent) {
+    // query textarea element
+    const textarea = document.getElementById(i.id);
+
+    if (textarea instanceof HTMLTextAreaElement) {
+      event.preventDefault();
+      const { selectionStart, selectionEnd } = textarea;
+      const selectedText = i.getSelectedText() || url.toString();
```

Comment:
If the return value of `getSelectedText` can be null or undefined, it is slightly clearer to use `??` instead of `||`.

Reply:
I actually used `||` specifically since `getSelectedText` can't return `null` or `undefined` but could return `""`. I wanted to fallback to the url in that case, but since the consensus seems to be that the smart formatting should not be applied if no text is selected this fallback will become obsolete.